### ISSUE TITLE
NIST source needs to be updated

### ISSUE THREAD

**Issue by arafey** [nhorman/rng-tools#69]
Hi,

The NIST entropy source doesn't seem to work anymore. I haven't looked too deeply but I noticed this: https://github.com/nhorman/rng-tools/blob/9fc873c5af0e392632e6b736938b811f7ca97251/rngd_nistbeacon.c#L65

According to the timestamp of the last pulse, version 1.0 hasn't been active since 2018. To use version 2.0, it should look like this:
```c
#define NIST_RECORD_URL "https://beacon.nist.gov/beacon/2.0/pulse/last"
``` 
It also looks like there are some significant protocol/format changes between 1.0 and 2.0:

## [version 1.0](https://beacon.nist.gov/1.0) @ 1545840420:

```xml
<record xmlns="http://beacon.nist.gov/record/0.1/">
<version>Version 1.0</version>
<frequency>60</frequency>
<timeStamp>1545840420</timeStamp>
<seedValue>
C2E5D63FA9B8276C1E4568BAE5B50C242856707519C295AEC71B99E72A0588CFEE2D79A7BC7ADD71D8BAC87FF753730D8DBDC171CE2DF6D0717C7E7AC45EE197
</seedValue>
<previousOutputValue>
2BF94CA75B69061C4440C606270AF37F2993782ED52FF80485BE221318465577A10D27AB93BEA19188BEFF14703BD723E356D4D5E9EEC420F1A64412A7A7FE6D
</previousOutputValue>
<signatureValue>
8D99A4E546A9DA0CA8F88F1C28EA411CF086125CE9B9728DDE0997E3C7EC8C32BA1CC764D745D372F79B5607E338003B3C1684CD1EA5972C06F1FB80AD363B77BF63E186C6E514DB39FFF12E190234FE12C4494D44D3A9BFE175AF04560F99DD1AA0724C5111ADC22268C6B8ADB4ECFCF0059266F673FA3F3EA02D404F505CCB6338808944E22C510871037993123C9932EBDF7802F45FBE280146121688BCBE08CE11DDE12B95CC57C028BDE2653C3612C754ACC2A7F8249F7E7EAC98C32354E6C06C9909C707A597AE7D99951B52F893F23167725347B5953E9416DE229F7E4B4361CC81D7D17A1DD779759A82CE48B4FF2774FFEE1AD33BA12493E2DA025D
</signatureValue>
<outputValue>
1A13A30ED048D48603F062993C69648C0FBF56DB5FE13FA3817511FFCD6AD1E7C81C7F82055DB20E18993678E9C71BBC1C4228C29C8320CE477DF1CE9945AF1E
</outputValue>
<statusCode>0</statusCode>
</record>
```

## [version 2.0](https://beacon.nist.gov/home) @ 1545840420

```json5
{
  "pulse" : {
    "uri" : "https://beacon.nist.gov/beacon/2.0/chain/1/pulse/220394",
    "version" : "Version 2.0",
    "cipherSuite" : 0,
    "period" : 60000,
    "certificateId" : "5501e3d72bc42f3b96e16de4dcadcb16768e109662bd16d667d5fd9aee585af31bbdc5dd4f53592276064b53dddd76c8f3604b2a41db6e09f78f82bb5d6569e7",
    "chainIndex" : 1,
    "pulseIndex" : 220394,
    "timeStamp" : "2018-12-26T16:07:00.000Z",
    "localRandomValue" : "5FF1E44E70C019C42C77FA72D5228A2E663416D0778BFAC826F6B4757B634B076C50ED2D5A3975CBAF237C211A027EDAFF3E241A885D69EAA7237E2744E6C1E2",
    "external" : {
      "sourceId" : "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
      "statusCode" : 0,
      "value" : "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
    },
    "listValues" : [ {
      "uri" : "https://beacon.nist.gov/beacon/2.0/chain/1/pulse/220393",
      "type" : "previous",
      "value" : "BA646CC4E7AE195D2C85E9D3AE9C9722B974F2134699D2493FA9E296C34995E8E471B329CB5F63235982CEE3395A749C618E61466847951D543ADC2FBAD23ECB"
    }, {
      "uri" : "https://beacon.nist.gov/beacon/2.0/chain/1/pulse/220387",
      "type" : "hour",
      "value" : "E75A5877169CC15220BCB11C8DA18159F14B880D85C5F3E9E462D010DC49BFCFE36D116D72C1D32A95AE8FFD9F0B6CE20DC073ED881BA36D5EF4687DC5B12328"
    }, {
      "uri" : "https://beacon.nist.gov/beacon/2.0/chain/1/pulse/219427",
      "type" : "day",
      "value" : "CDD24473B4427C3D3C856C66DF669444CE79D1262F94F4CC745E037AB781245A560E722514A62BEFF9ABE3B72EAFDF5EAE5A43EA806F5571B05EA04B8E7B02B7"
    }, {
      "uri" : "https://beacon.nist.gov/beacon/2.0/chain/1/pulse/183558",
      "type" : "month",
      "value" : "A9EDA202336C7DB1F05DE3BB24AAC1B54E98C9BD46CDF3D193FAE2BF4E0CD696AD6A743DFDF4DC48E5985BE329652E0A74816C7B69BBAF644FF0ACA352207FFB"
    }, {
      "uri" : "https://beacon.nist.gov/beacon/2.0/chain/1/pulse/1",
      "type" : "year",
      "value" : "7665F054F21B50DF62CD3E50AF8EB783E30D271B091DE051212D301E0E3D17FFCF0367DB41CFFD3C51E88BDE0B0621F49EB03435BC373D5D49480941A8B3547E"
    } ],
    "precommitmentValue" : "269908B840E79BE71991FFE62CEC4EBAEB3C050E93D71248CBB3E4358445FF0858D1D2CCF899A19B861C0C11CECCF16A0859AFD68E58481D4ADB1BE61F30E419",
    "statusCode" : 0,
    "signatureValue" : "17943D886DA8C7C24B9244BE5BD5DF281983D28CFFC8928846BC26529309C9724F6849F039591361DAF6B8DDAB6BC275CF86F448AF1800996889508D08D8AAAD19586E7A4B04FC4C97F1DA6D619EFAE2332150328C79C23BB9FE6A03E8FABDFF1AD66C5A8789D28AED4D25FF0FC5E88BE366280D7516A504EFD63706641828DDBF3C7082524F36E77EE9E07A9801D0C3BAD0646AA89DDDD8E2B4C0D7F8ED67664864B598E59ABF20CA8D761BB7B32B9A32698A22935D2C7127952625BB5580B2847FBBC8DFEF9039C4F5ACF12877E11121D031AED58217286F8DCF291C6E315773B42FF470B1AB587F787D44381F6E655DB903F1601B65AAC86BC2B7083AEEA9B3A27A5A208674056EFBB3C44629F333C810ADAD00E4FCFCE48E54F8FB7FC700598EA3E6497821736D24E5DA801A8B9DEC28A2B68E50FE13752270EB9CA7912B21EB6C104E78D105D0C0A635686B9A8CA26F87A1E63F0E411FD228F21B08BCD24660B305A4A42A9229154DE364FAB4DFF257A59DEA814034BF65C38A4C7AAEE79FEE5CC69010B1FE9759E23F192E218A19D9B8E95F6DD37D5D2F672E6CF0A0D457D9C619B1808274C2B0B2D3A3A7A8D8B1BB423FDDC56110784F2E0B7A23F065B56EC6E40234786DAB8C840E47811950331CBCFADEAD2EEE901D1C0A3A7E18D18A93089FC4E1CEFBA7571D2E47F10893D24BAD967FCA9DAEA67AD6B7F390AFC0",
    "outputValue" : "0A8863E03E200F694CBA50F0F9A009B078555FE637B07CA2C0A0E4D564080173787B26376C4762377A139D1BCAA916A10419504850EB7CF91552A17FDCAA0463"
  }
}
```

###### Source: [(NISTIR 8213 DRAFT) A Reference for Randomness Beacons: Format and Protocol Version 2](https://doi.org/10.6028/NIST.IR.8213-draft)

**Comment by nhorman** [comment]
yeah, it looks like they turned it off, which is unfortunate, since version 2.0 has been in beta since well before that.    I was waiting until the 2.0 schema was officially released before implementing it, but now I'm not sure what state its going to be left it.  I'll check with NIST to see if they plan on publishing, and either update rngd accordingly or deprecate the entire source.

Thanks!

**Comment by nhorman** [comment]
fixed by PR #87
